Kentico.Kontent.Delivery.Caching/DeliveryCacheManagerFactory.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Kentico.Kontent.Delivery.Abstractions;
+using Microsoft.Extensions.Caching.Distributed;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using System;
+using System.Linq;
+
+namespace Kentico.Kontent.Delivery.Caching
+{
+    /// <summary>
+    /// A factory class for <see cref="IDeliveryCacheManager"/>
+    /// </summary>
+    public class DeliveryCacheManagerFactory : IDeliveryCacheManagerFactory
+    {
+        private readonly IOptionsMonitor<DeliveryCacheManagerFactoryOptions> _optionsMonitor;
+        private readonly IServiceProvider _serviceProvider;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeliveryCacheManagerFactory"/> class
+        /// </summary>
+        /// <param name="optionsMonitor">A <see cref="DeliveryCacheManagerFactory"/> options</param>
+        /// <param name="serviceProvider">A <see cref="IServiceProvider"/> instance</param>
+        public DeliveryCacheManagerFactory(IOptionsMonitor<DeliveryCacheManagerFactoryOptions> optionsMonitor, IServiceProvider serviceProvider)
+        {
+            _optionsMonitor = optionsMonitor;
+            _serviceProvider = serviceProvider;
+        }
+
+        /// <summary>
+        /// Returns a named <see cref="IDeliveryCacheManager"/>.
+        /// </summary>
+        /// <param name="name">A name of <see cref="IDeliveryCacheManager"/> configuration</param>
+        /// <returns>The <see cref="IDeliveryCacheManager"/> instance that represents named cache manager</returns>
+        public IDeliveryCacheManager Get(string name)
+        {
+            if (name == null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            IDeliveryCacheManager manager = null;
+            var deliveryCacheManagerFactoryOptions = _optionsMonitor.Get(name);
+            var deliveryCacheOptions = deliveryCacheManagerFactoryOptions?.DeliveryCacheOptions.LastOrDefault()?.Invoke();
+            if (deliveryCacheOptions != null)
+            {
+                if (deliveryCacheOptions.CacheType == CacheTypeEnum.Memory)
+                {
+                    var memoryCache = _serviceProvider.GetService<IMemoryCache>();
+                    if (memoryCache != null)
+                    {
+                        manager = new MemoryCacheManager(memoryCache, Options.Create(deliveryCacheOptions));
+                    }
```

Comment:
If the memory cache isn't registered should you throw an exception that a required dependency is not yet registered?

Reply:
Thanks for the catch. I add a null check in the *CacheManagers